nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/PathHelper.cs

```diff
@@ -87,6 +87,55 @@ public static string[] GetAllDirectoriesToRoot(string initialDirectoryPath, stri
         return candidatePaths.ToArray();
     }
 
+    /// <summary>
+    /// Resolves the case of the file path in a case-insensitive manner. Returns null if the file path is not found. file path must be a full path inside the repoRootPath.
+    /// </summary>
+    /// <param name="filePath">The file path to resolve.</param>
+    /// <param name="repoRootPath">The root path of the repository.</param>
+    public static string? ResolveCaseInsensitivePathInsideRepoRoot(string filePath, string repoRootPath)
+    {
+        if (string.IsNullOrEmpty(filePath) || string.IsNullOrEmpty(repoRootPath))
+        {
+            return null; // Invalid input
+        }
+
+        // Normalize paths
+        var normalizedFilePath = filePath.FullyNormalizedRootedPath();
+        var normalizedRepoRoot = repoRootPath.FullyNormalizedRootedPath();
+
+        // Ensure the file path starts with the repo root path
+        if (!normalizedFilePath.StartsWith(normalizedRepoRoot, StringComparison.OrdinalIgnoreCase))
```

Comment:
Use `PathHelper.IsSubdirectoryOf()` here.  It should handle the case where `.StartsWith()` could return true, but it's not really a sub dir.  Example: `"/src/client-new/client-new.csproj".StartsWith("/src/client")` will return `true` because the strict string prefix matches, but it's not actually under the directory.

This would be a good unit test.

Reply:
Actually, `IsSubdirectoryOf()` won't work with files, it's just directories, so you could change this one line to be:

`if (!normalizedFilePath.StartsWith(normalizedRepoRoot + "/", StringComparison.OrdinalIgnoreCase))`

That will ensure it's actually a directory.